File: src/main/java/domain/placement.java
package domain;

import java.time.Instant;

public final class placement {
        private final long id;
        private final long containerID;
        private final long slotID;
        private final long sampleID;
        private final Instant placedAt;
        private String ownerUsername;

        public placement(long id, long sampleID, long containerID, long slotId, String ownerUsername) {
            this.id = id;
            this.containerID = containerID;
            this.slotID = slotId;
            this.sampleID = sampleID;
            this.placedAt = Instant.now();
            this.ownerUsername = ownerUsername;
        }

        public long getID() { return id; }
        public long getContainerID() { return containerID; }
        public long getSlotID() { return slotID; }
    public long getSampleID(){return sampleID;}
        public Instant getPlacedAt() { return placedAt; }
        public String getOwnerUsername() { return ownerUsername; }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (!(o instanceof placement)) return false;
            placement p = (placement) o;
            return id == p.id;
        }
        @Override
        public int hashCode() {
            return Long.hashCode(id);
        }
    }

File: src/main/kotlin/Domain/Placement.kt
package Domain




import java.time.Instant

class Placement(
    val id: Long,
    val sampleId: Long,
    val containerId: Long,
    val slotId: Long,
    val ownerUsername: String = "SYSTEM"
) {
    val placedAt: Instant = Instant.now()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Placement) return false
        return id == other.id
    }

    override fun hashCode(): Int = id.hashCode()
}